diff -Naur upower-0.9.5_orig/src/linux/up-device-supply.c upower-0.9.5/src/linux/up-device-supply.c
--- upower-0.9.5_orig/src/linux/up-device-supply.c	2010-07-02 16:04:42.000000000 +0300
+++ upower-0.9.5/src/linux/up-device-supply.c	2011-04-20 12:01:31.787624219 +0300
@@ -491,11 +491,14 @@
 		energy_full_design = sysfs_get_double (native_path, "energy_full_design") / 1000000.0;
 
 		/* convert charge to energy */
-		if (energy == 0) {
-			energy_full = sysfs_get_double (native_path, "charge_full") / 1000000.0;
-			energy_full_design = sysfs_get_double (native_path, "charge_full_design") / 1000000.0;
-			energy_full *= voltage_design;
-			energy_full_design *= voltage_design;
+		if ((energy == 0) || (energy_full == 0)) {
+			energy_full = sysfs_get_double (native_path, "charge_full");
+			energy_full = (energy_full * voltage_design) / 1000000.0;
+			supply->priv->coldplug_units = UP_DEVICE_SUPPLY_COLDPLUG_UNITS_CHARGE;
+		}
+		if ((energy == 0) || (energy_full_design == 0)) {
+			energy_full_design = sysfs_get_double (native_path, "charge_full_design");
+			energy_full_design = (energy_full_design * voltage_design) / 1000000.0;
 			supply->priv->coldplug_units = UP_DEVICE_SUPPLY_COLDPLUG_UNITS_CHARGE;
 		}
 
