Signed-off-by: Ruediger Gad <r.c.g@gmx.de>
---
diff --git a/src/notify-upower.sh b/src/notify-upower.sh
index d88ae5e..87b6311 100755
--- a/src/notify-upower.sh
+++ b/src/notify-upower.sh
@@ -1,5 +1,5 @@
 #!/bin/bash
-[ "$1" = "post" ] && exec /usr/bin/dbus-send	\
+[ "$1" = "post" ] && exec /bin/dbus-send	\
 	--system --type=signal			\
 	--dest=org.freedesktop.UPower		\
 	/org/freedesktop/UPower			\
